UnityGLTF/Assets/UnityGLTF/Scripts/Extensions/GLTFSceneExporter.Animation.cs

```diff
@@ -0,0 +1,335 @@
+				values => this.ExportData(values));
+		}
+
+		private static Quaternion CreateNormalizedQuaternion(float x, float y, float z, float w)
+		{
+			var factor = 1.0f / Mathf.Sqrt(x * x + y * y + z * z + w * w);
+			return new Quaternion(x * factor, y * factor, z * factor, w * factor);
+		}
+
+		private AnimationSampler ExportAnimationSamplerRotation(AnimationCurve curveX, AnimationCurve curveY, AnimationCurve curveZ, AnimationCurve curveW)
+		{
+			return this.ExportAnimationSampler(
+				new[] { curveX, curveY, curveZ, curveW },
+				keyIndex => GetRightHandedQuaternion(new Quaternion(curveX.keys[keyIndex].inTangent, curveY.keys[keyIndex].inTangent, curveZ.keys[keyIndex].inTangent, curveW.keys[keyIndex].inTangent)),
+				keyIndex => GetRightHandedQuaternion(CreateNormalizedQuaternion(curveX.keys[keyIndex].value, curveY.keys[keyIndex].value, curveZ.keys[keyIndex].value, curveW.keys[keyIndex].value)),
+				keyIndex => GetRightHandedQuaternion(new Quaternion(curveX.keys[keyIndex].outTangent, curveY.keys[keyIndex].outTangent, curveZ.keys[keyIndex].outTangent, curveW.keys[keyIndex].outTangent)),
+				time => GetRightHandedQuaternion(CreateNormalizedQuaternion(curveX.Evaluate(time), curveY.Evaluate(time), curveZ.Evaluate(time), curveW.Evaluate(time))),
+				values => this.ExportData(values));
+		}
+
+		private AnimationSampler ExportAnimationSamplerScale(AnimationCurve curveX, AnimationCurve curveY, AnimationCurve curveZ)
+		{
+			return this.ExportAnimationSampler(
+				new[] { curveX, curveY, curveZ },
+				keyIndex => new Vector3(curveX.keys[keyIndex].inTangent, curveY.keys[keyIndex].inTangent, curveZ.keys[keyIndex].inTangent),
+				keyIndex => new Vector3(curveX.keys[keyIndex].value, curveY.keys[keyIndex].value, curveZ.keys[keyIndex].value),
+				keyIndex => new Vector3(curveX.keys[keyIndex].outTangent, curveY.keys[keyIndex].outTangent, curveZ.keys[keyIndex].outTangent),
+				time => new Vector3(curveX.Evaluate(time), curveY.Evaluate(time), curveZ.Evaluate(time)),
+				values => this.ExportData(values));
+		}
+
+		private AnimationSampler ExportAnimationSamplerWeight(IEnumerable<AnimationCurve> weightCurves)
+		{
+			return this.ExportAnimationSampler(
+				weightCurves,
+				keyIndex => weightCurves.Select(curve => curve.keys[keyIndex].inTangent / 100),
+				keyIndex => weightCurves.Select(curve => curve.keys[keyIndex].value / 100),
+				keyIndex => weightCurves.Select(curve => curve.keys[keyIndex].outTangent / 100),
+				time => weightCurves.Select(curve => curve.Evaluate(time) / 100),
+				values => this.ExportData(values.SelectMany(value => value)));
+		}
+
+		private AnimationSampler ExportAnimationSampler<T>(IEnumerable<AnimationCurve> curves, Func<int, T> getInTangent, Func<int, T> getValue, Func<int, T> getOutTangent, Func<float, T> evaluate, Func<IEnumerable<T>, AccessorId> exportData)
+		{
+			//decide if we should include baking animations or not
+			if (/*bakeAnimations is False &&*/ CanExportCurvesAsSpline(curves))
```

Comment:
This baking code can use a bit of work. We should at least not write the same output value if they are close enough. I also just picked an arbitrary 30 fps to bake the animation which might or might not be right.